api/src/main/java/org/openmrs/Order.java

```diff
@@ -462,7 +501,7 @@ public boolean isDiscontinued(Date checkDate) {
 		if (checkDate == null) {
 			checkDate = new Date();
 		}
-		if (dateActivated == null || !isStarted(checkDate) || dateStopped == null) {
+		if (dateActivated == null || !isActivated(checkDate) || dateStopped == null) {
```

Comment:
isActivated() already checks if dateActivated  == null, so you don't need to check if it is null again here